src/buffer.c

```diff
@@ -114,7 +114,15 @@ bool buffer_expand_capacity(buffer_T* buffer) {
 bool buffer_expand_if_needed(buffer_T* buffer, const size_t required_length) {
   if (buffer_has_capacity(buffer, required_length)) { return true; }
 
-  return buffer_resize(buffer, buffer->capacity + (required_length * 2));
+  bool fits_in_doubled_buffer = required_length < buffer->capacity;
```

Comment:
Thanks for working on this, @timkaechele!

Shouldn't this be `buffer->capacity * 2`? 

```suggestion
  bool fits_in_doubled_buffer = required_length < buffer->capacity * 2;
```

If not, then I guess we should rename the variable from `fits_in_doubled_buffer` to something else, or am I missing something?

Reply:
Ah, good question, but it should be only the the single buffer capacity.

Reasoning: If you resize the buffer because it is nearly full and you plan to double the capacity you only get `buffer->capacity` more space after the resize. Hence it is `buffer->capacity`.

But I admit the name of the variable can be confusing, I couldn't come up with something significantly better though.

Candidates such as `fits_in_additional_capacity_after_doubling` felt a bit clumsy.

Reply:
Ah right, I see the trouble, again.

I just renamed this this `should_double_capacity`, thanks!